### PR TITLE
Feature/milestone2

### PR DESCRIPTION
**Pull request by jessicafarias**
# **This is the second milestone of the calculator project.**
### App component
-  App should render the Display and ButtonPanel component. 

### Display component
 - Accept the result of the calculation as an incoming prop.
 - The result prop requires a string with props-type
 - The default value of the result prop is 0 ( using defaultProps)

### Button component
 - Accept a button name as a prop.
 - The button name prop should is String.

### ButtonPanel component
 - Rendering the calculator panel with all the buttons in the following groups: (I'm using divs to divide buttons into groups)
 - Group 1: AC, +/-, %, ÷
 - Group 2: 7, 8, 9, X
 - Group 3: 4, 5, 6, -
 - Group 4: 1, 2, 3, +
 - Group 5: 0, ., =

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React, { PureComponent } from 'react';
+import Display from './Display';
+import ButtonPanel from './ButtonPanel';
+
+class App extends PureComponent {
+  render() {
+    return (
+      <div>
```
Comment (Forison) [inline src/components/App.js]: - Use a fragment in other to create an efficient render and no creation of extra node. `<></>`

File: src/components/Button.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Button = props => {
```
Comment (Forison) [inline src/components/Button.js]: **[optional]**
we can equally destructure at the argument level like this `const Button = ({name}) => {`

File: src/components/Display.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  );
+};
+
+Display.propTypes = {
+  result: PropTypes.string,
+
+};
+
+Display.defaultProps = {
+  result: 0,
```
Comment (Forison) [inline src/components/Display.js]: Change the result value to string
result: '0'

File: src/components/ButtonPanel.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import Button from './Button';
+
+const ButtonPanel = () => (
+  <div className="calculator-base">
+    <div>
```
Comment (Forison) [inline src/components/ButtonPanel.js]: **[OPTIONAL]**
You could store the button name list in an array, loop over the array containing the button names and render button each time you loop over the button name array, it is much more cleaner than what we have now.

**Review by Forison — CHANGES_REQUESTED** [review]
## Status::Changes Required❌

# 👋 @jessicafarias,
  
#### Good job done on this one so far, however, there are a few issues to address to get this project approved.
 
- App Component 

* [x] App.jsx  has children components wrapped in [React.Fragment](https://reactjs.org/docs/fragments.html#short-syntax) short syntax ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/app/ReactFragment.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896092/))

- Display Component

* [x] its result prop should be a String with default value = '0' ([should be String screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/display/validateAsString.png)) ([default props screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/display/DefaultProp.png)). currently typeOf result would return number instead of string.

- check my comments below👇
#### That is it kindly make the changes and submit a re-review👷.
#### Happy coding👨‍💻
Addo Forison

**Comment by jessicafarias** [comment]
Thanks alot for this helpful review.

**Review by rOluochKe — APPROVED** [review]
Hello @jessicafarias 
### Good work in making the changes, the project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next one.
- Happy coding!

**Pull request merged**